lib/Parser/RegexParser.cpp

```diff
@@ -2199,6 +2224,7 @@ namespace UnifiedRegex
                         Assert(!unicodeFlagPresent);
                         Fail(JSERR_RegExpBadRange);
```

Comment:
I'm concerned about this comment. Mostly, I'm not sure if it's accurate to the current logic, and I wonder if at some point this codepath stopped checking the right thing, or just stopped agreeing with the comment, or whether the comment is correct and needs more explaining. I don't think your change affects the truth of this comment, but it's worth examining a bit.

We want to throw an error for an ordinary out of range, which this codepath does.

We also need to handle the case of surrogate code points, if applicable. The assert below the comment producing no runtime failures -- assuming you ran RegexTestCollateral on a debug build -- indicates that it is not possible to reach this point if /u is specified. We know we can't get here with /u and prevprevWasACharSetAndPartOfRange because we would have thrown SyntaxError earlier. As the assert at the beginning of the function `Assert(containsSurrogates ? unicodeFlagPresent : true);` if the input string contains surrogates -- which is already known on the codepath where this function is called (since containsSurrogates is a template parameter) -- then we must have /u as well.

According to the Assert, getting here means it's not possible for all of these checks to be true when `!unicodeFlagPresent`

```cpp
            if (codePointToSet != INVALID_CODEPOINT)
            {
                if (pendingRangeStart != INVALID_CODEPOINT)
                {
                    if (pendingRangeStart > pendingCodePoint)
```

Reply:
In pass0 it will Fail if there is an invalid non-surrogate range, so I assume the comment is because in pass1,
`pendingRangeStart > pendingCodePoint ` 
is true only when there is a range containing surrogates